### PR TITLE
Better path handling

### PR DESCRIPTION
**Pull request by rlupton20**
There are places (#60) in `floki` where paths aren't dealt with correctly - there is no guarantee that a path can be utf-8 decoded, but we assume it always can.

This MR uses the correct types with paths, and tidies up some of the code surrounding them. Even if in practice most paths are valid utf-8, this change makes the code more correct, and clearer, as well as avoiding a few unneeded allocations. Correcter, cleaner, faster!

### CODE REVIEW

File: src/command.rs [hunk 1]
```diff
@@ -81,7 +82,7 @@ impl DockerCommandBuilder {
             .args(&["--name", &self.name])
             .args(&self.build_volume_switches())
             .args(&self.build_environment_switches())
-            .args(&self.build_docker_switches())
+            .args(self.build_docker_switches())
```
Comment (maxdymond) [inline src/command.rs]: Just seems weird that the other functions here are using & but this one isn't.

**Comment by maxdymond** [comment]
Honestly not much to say here as the comments are pretty clear and the commits are small enough to review individually. Happy to merge.

File: src/command.rs (hunk 1 above)
Comment (rlupton20) [reply src/command.rs]: The inner function returns a reference, so there is an implicit `&`, but it does suggests there may be simplifications to be had in the other inner functions. I'll take a look.

Comment (rlupton20) [reply src/command.rs]: Ok, I think there are some further simplifications to be had here, but I'm going to address them separately.

**Pull request merged**